GPU/GLES/GLES_GPU.cpp

```diff
@@ -463,6 +466,94 @@ GLES_GPU::~GLES_GPU() {
 	glstate.SetVSyncInterval(0);
 }
 
+// Take the raw GL extension and versioning data and turn into feature flags.
+void GLES_GPU::CheckGPUFeatures() {
+	u32 features = 0;
+	if (gl_extensions.ARB_blend_func_extended /*|| gl_extensions.EXT_blend_func_extended*/) {
+		if (gl_extensions.gpuVendor == GPU_VENDOR_INTEL || !gl_extensions.VersionGEThan(3, 0, 0)) {
+			// Don't use this extension to off on sub 3.0 OpenGL versions as it does not seem reliable
+			// Also on Intel, see https://github.com/hrydgard/ppsspp/issues/4867
+		} else {
+			features |= GPU_SUPPORTS_DUALSOURCE_BLEND;
+		}
+	}
+
+	if (gl_extensions.IsGLES) {
+		if (gl_extensions.GLES3)
+			features |= GPU_SUPPORTS_GLSL_ES_300;
+	} else {
+		if (gl_extensions.VersionGEThan(3, 3, 0))
+			features |= GPU_SUPPORTS_GLSL_330;
+	}
+
+	// Framebuffer fetch appears to be buggy at least on Tegra 3 devices.  So we blacklist it.
+	// Tales of Destiny 2 has been reported to display green.
+	if (gl_extensions.EXT_shader_framebuffer_fetch || gl_extensions.NV_shader_framebuffer_fetch || gl_extensions.ARM_shader_framebuffer_fetch) {
+		features |= GPU_SUPPORTS_ANY_FRAMEBUFFER_FETCH;
+		// Blacklist Tegra 3, doesn't work very well.
+		if (strstr(gl_extensions.model, "NVIDIA Tegra 3") != 0) {
+			features &= ~GPU_SUPPORTS_ANY_FRAMEBUFFER_FETCH;
```

Comment:
@hrydgard So here is for blacklist for my phone and @shinemu phone's code (PowerVR SGX 540) ?